go/kbfs/kbfsedits/tlf_history.go

```diff
@@ -386,6 +406,56 @@ func (r *recomputer) processNotification(
 		r.fileEvents[eventFilename] = fileEvent{delete: true}
 	case NotificationDelete:
 		r.fileEvents[eventFilename] = fileEvent{delete: true}
+
+		// We only care about files, so skip dir and sym creates.
+		if notification.FileType != EntryTypeFile {
+			return false
+		}
+
+		// Ignore macOS dotfiles.
+		if ignoreFile(filename) {
+			return false
+		}
+
+		wn, ok := r.byWriter[writer]
+		if !ok {
+			wn = &writerNotifications{writer, nil, nil}
+			r.byWriter[writer] = wn
+		}
+
+		if len(wn.deletes) == maxDeletesPerWriter {
+			// We don't need any more deletes, but we
+			// should continue looking for more edit notifications.
+			return false
+		}
+
+		if hasEvent && event.delete {
+			// It's already been deleted, no need to track it further.
```

Comment:
why would a file have multiple delete events?

Reply:
It could be created and deleted multiple times.  The edit history tracks files simply by their names, not by block ID or anything like the CR does.